plugins/newsql/newsql_evbuffer.c

```diff
@@ -900,19 +900,31 @@ static void rd_payload(int dummyfd, short what, void *arg)
     process_newsql_payload(appdata, query);
 }
 
+/* 10-second in-process cache TTL (does not affect idle connections or connections held by sockpool) */
+int gbl_inproc_conn_ttl = 10;
 static void rd_hdr(int dummyfd, short what, void *arg)
 {
     struct newsqlheader hdr;
     struct newsql_appdata_evbuffer *appdata = arg;
+    struct sqlclntstate *clnt = &appdata->clnt;
+    struct timeval idle_evbuffer_time = {
+        .tv_sec = gbl_inproc_conn_ttl,
+        .tv_usec = 0
+    };
+
     if (evbuffer_get_length(appdata->rd_buf) >= sizeof(struct newsqlheader)) {
         goto hdr;
     }
+    if (dummyfd >= 0 && (what & EV_TIMEOUT) && (clnt->in_local_cache && clnt->state == CONNECTION_RESET)) {
```

Comment:
Curious - when would connection be in local cache, but not be CONNECTION_RESET?

Reply:
There shouldn't be such a case. Let me remove the check.

Reply:
Do we need 2 separate places to track this state then?

Reply:
I changed it to only check `clnt->in_local_cache`.